From 9e59f4ba56c71b1655f82915806a42c22df9dd18 Mon Sep 17 00:00:00 2001
From: Timo Herbrecher <t.herbrecher@gateware.de>
Date: Fri, 12 Oct 2018 11:31:27 +0200
Subject: [PATCH] sdhci: Supress data interrupt error message

sdhci controller fires error message when a data complete interrupt is
seen while no data transfer is processed. Supress error message because
the interrupt does not harm driver function.

Signed-off-by: Timo Herbrecher <t.herbrecher@gateware.de>
---
 drivers/mmc/host/sdhci.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/mmc/host/sdhci.c b/drivers/mmc/host/sdhci.c
index b48565e..0493ba8 100644
--- a/drivers/mmc/host/sdhci.c
+++ b/drivers/mmc/host/sdhci.c
@@ -2451,10 +2451,11 @@ static void sdhci_data_irq(struct sdhci_host *host, u32 intmask)
 			}
 		}
 
-		pr_err("%s: Got data interrupt 0x%08x even "
-			"though no data operation was in progress.\n",
-			mmc_hostname(host->mmc), (unsigned)intmask);
-		sdhci_dumpregs(host);
+		if (intmask & ~SDHCI_INT_DATA_END) {
+			pr_err("%s: Got data interrupt 0x%08x even though no data operation was in progress.\n",
+			       mmc_hostname(host->mmc), (unsigned)intmask);
+			sdhci_dumpregs(host);
+		};
 
 		return;
 	}
-- 
1.9.1

